### PR TITLE
Soil Conditions file refactoring

### PR DESCRIPTION
**Pull request by dearpriyankasa**

### CODE REVIEW

**Comment by mikahpinegar** [comment]
unselecting the chips is broken for drainage class and flooding frequency

File: src/components/Location/SoilCondition/SoilCondition.js [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+              <Typography variant="body1" display="inline">
+                NO
+              </Typography>
+              <Switch
+                checked={tilingCheck}
+                onChange={(e) => {
+                  let soilDrainCopy = soilData.Drainage_Class;
+
+                  let drainSet = new Set(soilDrainCopy);
+                  if (e.target.checked) {
```
Comment (mikahpinegar) [inline src/components/Location/SoilCondition/SoilCondition.js:211]: i think all this logic is broken

**Review by mikahpinegar — CHANGES_REQUESTED** [review]

**Comment by mikahpinegar** [comment]
i think only one chip should be selectable for flooding frequency and drainage class but i'll need to get clarification. for now just make it so the chips can be unselected and the broken logic above works properly

File: src/components/Location/SoilCondition/SoilCondition.js (hunk 1 above)
Comment (dearpriyankasa) [reply src/components/Location/SoilCondition/SoilCondition.js:211]: Okay, I'll take a look

**Review by mikahpinegar — APPROVED** [review]

**Pull request merged**